src/v/coproc/reconciliation_backend.cc

```diff
@@ -88,7 +130,9 @@ reconciliation_backend::process_events_for_ntp(
         vlog(coproclog.trace, "executing ntp: {} op: {}", d.ntp, d);
         auto err = co_await process_update(d);
         vlog(coproclog.info, "partition operation {} result {}", d, err);
-        if (err == errc::partition_not_exists) {
+        if (
+          d.type == update_t::op_type::add_non_replicable
+          && err == errc::partition_not_exists) {
```

Comment:
Confusingly if we observe the following op types:
```
add -> add materialized -> update
```
Then on restart, since its impossible to synchronize around the update and add events (because they are handled by `controller_backend`) when processing `add_materialized` occurs there's a possibility the source topic doesn't exist because the `controller_backend` already moved it across shards or onto another node.

There is an order dependency between the processing of the controller log between this class and `controller_backend`. Moreso they can be processing the controller log at totally different offset. Maybe it would be cleaner to put all events event add / delete coprocessor ones in the controller log, but aside from that a solution to the aforementioned problem would be to just verify if this event can be ignored by checking if the partition assignments for the source topic and materialized topics (found in topics table) and equal. If they aren't the processing of the op_t can be ignored, safely assuming there is a future `update` request incoming.

For example:
```
add (always ignored by reconciliation_backend) -> add materialized (sees non matching partition assignments, skips) -> update (attempts to create materialized partition)
```
and things continue to work as normal, since an update in any case always creates a materialized partition.

Reply:
To add onto this, when processing for any op_t returns `partition_not_exists` forward progress cannot be made until the op_t is processed. Possibilities for infinite looping could occur here